File: samples/client/petstore/java/src/main/java/org/openapijsonschematools/client/schemas/validation/KeywordValidator.java
package org.openapijsonschematools.client.schemas.validation;

import org.openapijsonschematools.client.exceptions.ValidationException;
import org.checkerframework.checker.nullness.qual.Nullable;

import java.util.ArrayList;
import java.util.List;

public interface KeywordValidator {
    @Nullable PathToSchemasMap validate(
        JsonSchema schema,
        @Nullable Object arg,
        ValidationMetadata validationMetadata,
        @Nullable List<PathToSchemasMap> containsPathToSchemas
    ) throws ValidationException;

    default List<PathToSchemasMap> getContainsPathToSchemas(
        @Nullable Object arg,
        ValidationMetadata validationMetadata
    ) {
        return new ArrayList<>();
    }
}

File: samples/client/petstore/kotlin/src/main/kotlin/org/openapijsonschematools/client/schemas/validation/KeywordValidator.kt
package org.openapijsonschematools.client.schemas.validation;

import org.openapijsonschematools.client.exceptions.ValidationException;
import org.checkerframework.checker.nullness.qual.Nullable;

@FunctionalInterface
public interface KeywordValidator {
    @Nullable PathToSchemasMap validate(
        ValidationData data
    ) throws ValidationException;
}